Draft: Add a watermark widget

alxflam opened the pull request:
#### What it does
Adds a watermark widget which, similar to the one in VSCode, displays useful commands and their keybindings.
Closes #10804.

Marked this as a draft for now due to some open topics:
- Is the implementation approach taken fine? (adding a new contribution point for watermark commands and moving existing keybinding rendering logic from the keymaps package into core to enable a watermark widget already in core instead of in a separate package - see #10804 for earlier discussion)
-  Which commands should be shown as watermarks? (for now, i only used a subset of the same commands VSCode shows for demonstration - i'd simply use the same commands VSCode displays)

![watermark](https://user-images.githubusercontent.com/23746778/159093033-189ac0b9-2566-48ed-a5b9-d278739dab4e.png)

#### How to test
- Verify watermark widget is shown when there's no open editor

#### Review checklist

- [X] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

vince-fugnitto reviewed (commented):
@alxflam thank you for your contribution and interest in implementing the watermark :+1: 

I noticed a couple of issues with a high-level review:

1. It might be overkill to move all the rendering logic of keybindings to core, all you really need is the CSS styling. The watermark should not need to know about doing fuzzy matching (highlights when searching) or any specifics of the keybindings-widget. The changeset should then be much smaller.
2. The keybindings-widget no longer works from what I tested, when searching no results exist:
![keybindings-search-bug](https://user-images.githubusercontent.com/40359487/159258471-5520a476-7a2c-4f07-baa5-62fac810de34.gif)
3. On electron I noticed that a command does not show the `label`, instead we get the `id`:

![image](https://user-images.githubusercontent.com/40359487/159258641-0c27016b-d964-4324-accd-669c34d367b2.png)

alxflam commented:
@vince-fugnitto thanks for your feedback. I removed the code in KeybindingUtil to the necessary minimum and moved everything back to the keymaps package (also solves the issue in the keybindings widget). The 'Open File' command does not have a label, only a dialogLabel - seems this caused the issue on electron (for now i added a fallback to dialogLabel in case label is not set).

Current state is still rather rough, so no in-depth review required.

vince-fugnitto commented:
@alxflam I think the general idea is fine if you want to continue, cleanup and finalize the pull-request. We can perform a more in depth review of the changes when it is finalized (it will likely go through further iterations so don't worry about converting it from a draft).

alxflam commented:
@vince-fugnitto thanks, i'll provide an update

JonasHelming commented:
@alxflam : What is the current state of this PR, is this ready for another review? If so, could you resolve the conflicts, please?

chakraborty-slx commented:
@alxflam are you looking into this?

alxflam commented:
@JonasHelming @chakraborty-slx It might take a while until i can get another look at this, if someone else wants to take over feel free to do so and reuse my commits.

vince-fugnitto commented:
@alxflam friendly note that we might close the pull-request soon for the moment to keep the list of open pull-requests more manageable but we can always re-open and re-visit the pull-request in the future when needed and when updated.
